### ISSUE TITLE
[java] LocalVariableCouldBeFinal on lambda parameters

### ISSUE THREAD

**Issue by jsotuyod** [pmd/pmd#1103]
Lambda parameters are not intrinsically final, but more often than not, lambdas refrain from declaring / assigning variables, meaning the final keyword is mostly redundant.

I think we should whitelist these cases for the rule, at least for the scenario where we use a 1-liner (when the lambda is defined in a block this may not be true).

```java
stream.filter(v -> v > 0); // no need to declare v as final

stream.filter(v -> { // maybe require v to be final? multiple statements follow 
  if (v > 0) {
    return true;
  }

  return goCheckSomethingElse(v);
});
```

**Comment by oowekyala** [comment]
This would probably only be acceptable with a language version over [Java 11](http://openjdk.java.net/jeps/323), which allows lambda parameters to be declared as `final var`. Needing to declare the type of a lambda parameter explicitly is more often a nuisance than not imo. A somewhat extreme example would be this lambda from the designer
```java
.filterMap(t -> {
    if (t.isSuccess()) {
        return Optional.of(t.get());
    } else {
        t.getFailure().printStackTrace();
        return Optional.empty();
    }
})
```

Which would need to be rewritten as

```java
.filterMap((final Try<StyleSpans<Collection<String>>> t) -> {
    ...
})
```

**Comment by oowekyala** [comment]
Since PMD 7 we don't report lambda parameters anymore as the grammar has been changed. I added a test to make sure of this.
